src/app/fdctl/run/tiles/fd_repair.c

```diff
@@ -0,0 +1,656 @@
+
+static void
+repair_shred_deliver_fail( fd_pubkey_t const * id FD_PARAM_UNUSED, 
+                           ulong               slot, 
+                           uint                shred_index,
+                           void *              arg FD_PARAM_UNUSED,
+                           int                 reason ) {
+  FD_LOG_WARNING(( "repair failed to get shred - slot: %lu, shred_index: %u, reason: %u", slot, shred_index, reason ));
+}
+
+static void
+before_frag( void * _ctx FD_PARAM_UNUSED,
+             ulong  in_idx,
+             ulong  seq  FD_PARAM_UNUSED,
+             ulong  sig,
+             int *  opt_filter ) {
+
+  if( FD_LIKELY( in_idx==NET_IN_IDX ) ) {
+    *opt_filter = fd_disco_netmux_sig_proto( sig )!=DST_PROTO_REPAIR;
+  }
+}
+
+static void
+during_frag( void * _ctx,
+             ulong  in_idx,
+             ulong  seq FD_PARAM_UNUSED,
+             ulong  sig,
+             ulong  chunk,
+             ulong  sz,
+             int *  opt_filter ) {
+
+  fd_repair_tile_ctx_t * ctx = (fd_repair_tile_ctx_t *)_ctx;
+  uchar const * dcache_entry;
+  if( FD_UNLIKELY( in_idx==CONTACT_IN_IDX ) ) {
+    if( FD_UNLIKELY( chunk<ctx->contact_in_chunk0 || chunk>ctx->contact_in_wmark ) ) {
+      FD_LOG_WARNING(( "chunk %lu %lu corrupt, not in range [%lu,%lu]", chunk, sz,
+            ctx->contact_in_chunk0, ctx->contact_in_wmark ));
+      *opt_filter = 1;
+      return;
+    }
+    dcache_entry = fd_chunk_to_laddr_const( ctx->contact_in_mem, chunk );
+
+  } else if( FD_UNLIKELY( in_idx==STAKE_IN_IDX ) ) {
+    if( FD_UNLIKELY( chunk<ctx->stake_weights_in_chunk0 || chunk>ctx->stake_weights_in_wmark ) ) {
+      FD_LOG_WARNING(( "chunk %lu %lu corrupt, not in range [%lu,%lu]", chunk, sz,
+            ctx->stake_weights_in_chunk0, ctx->stake_weights_in_wmark ));
+      *opt_filter = 1;
+      return;
+    }
+
+    dcache_entry = fd_chunk_to_laddr_const( ctx->stake_weights_in_mem, chunk );
+    fd_stake_ci_stake_msg_init( ctx->stake_ci, dcache_entry );
+    return;
+
+  } else if( FD_UNLIKELY( in_idx==STORE_IN_IDX ) ) {
+    if( FD_UNLIKELY( chunk<ctx->repair_req_in_chunk0 || chunk>ctx->repair_req_in_wmark ) ) {
+      FD_LOG_WARNING(( "chunk %lu %lu corrupt, not in range [%lu,%lu]", chunk, sz,
+            ctx->repair_req_in_chunk0, ctx->repair_req_in_wmark ));
+      *opt_filter = 1;
+      return;
+    }
+
+    dcache_entry = fd_chunk_to_laddr_const( ctx->repair_req_in_mem, chunk );
+
+  } else if ( FD_LIKELY( in_idx == NET_IN_IDX ) ) {
+    if( FD_UNLIKELY( chunk<ctx->net_in_chunk0 || chunk>ctx->net_in_wmark || sz>FD_NET_MTU ) ) {
+      FD_LOG_WARNING(( "chunk %lu %lu corrupt, not in range [%lu,%lu]", chunk, sz, ctx->net_in_chunk0, ctx->net_in_wmark ));
+      *opt_filter = 1;
+      return;
+    }
+
+    dcache_entry = fd_chunk_to_laddr_const( ctx->net_in_mem, chunk );
+    ulong  hdr_sz = fd_disco_netmux_sig_hdr_sz( sig );
+    FD_TEST( hdr_sz < sz ); /* Should be ensured by the net tile */
```

Comment:
This check is not useful here since the bytes can be torn (it can pass and later be invalid anyway).